src/ceph-volume/ceph_volume/tests/devices/lvm/test_batch.py

```diff
@@ -65,7 +65,7 @@ def test_disjoint_device_lists(self, factory):
         b._filter_devices()
         with pytest.raises(Exception) as disjoint_ex:
             b._ensure_disjoint_device_lists()
-        assert 'Device lists are not disjoint' in str(disjoint_ex.value)
+            assert 'Device lists are not disjoint' in str(disjoint_ex.value)
```

Comment:
this change looks wrong to me

Reply:
care to elaborate?

`_ensure_disjoint_device_lists()` raises and Exception when called with overlapping device lists. The assert checks for the certain message in the assertion.

Reply:
https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Reply:
> https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Not actually an explanation. So turns out with, being just syntactic sugar doesn't open a new scope. In other words either indentation is fine.

Reply:
actually no:

```python
(venv) ➜  foo python
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pytest
>>> with pytest.raises(Exception) as ex:
...     raise Exception("foo")
...     assert 'bar' in str(ex.value)
# passes
```

problem is, as soon as we leave the indented block, the assert is no longer executed.  The scope of variables in Python is as bad as you can imagine, the control flow of statements is not as bad.

Reply:
oh right...syntactic sugar was the hint :)